src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
+
+    return next.handle().pipe(
+      map((data) => {
+        const response = {
+          isSuccess: statusCode >= 200 && statusCode < 300 ? true : false,
+          statusCode: statusCode,
+          error: data && data.message ? data.message : '',
+          message: data && data.message ? data.message : '',
+          data: data && data.message ? { ...data, message: undefined } : data,
+        };
+        this.logger.debug(
+          `Response: [${JSON.stringify(
+            response,
+          )}] || User Details: [${JSON.stringify(userDetail)}]`,
+        );
+
+        return response;
+      }),
+    );
+  }
```

Comment:
PullProofAI

Consider refactoring the mapping of response data for better readability and maintainability.
```typescript
map((data) => {
  const isSuccess = statusCode >= 200 && statusCode < 300;
  const error = data?.message || '';
  const message = data?.message || '';
  const responseData = data?.message ? { ...data, message: undefined } : data;

  const response = {
    isSuccess,
    statusCode,
    error,
    message,
    data: responseData,
  };

  this.logger.debug(`Response: [${JSON.stringify(response)}] || User Details: [${JSON.stringify(userDetail)}]`);

  return response;
}),
```
Splitting the logic into separate variables can improve code readability and make it easier to understand the structure of the response object.
```


<!-- This is an auto-generated comment by PullProofAI -->